fuse_core/include/fuse_core/throttled_callback.h

```diff
@@ -179,8 +217,60 @@ class ThrottledCallback
     }
   }
 
+  Callback keep_callback_;         //!< The callback to call when kept, i.e. not dropped
+  Callback drop_callback_;         //!< The callback to call when dropped because of throttling
+  ros::Duration throttle_period_;  //!< The throttling period duration in seconds
+  bool use_wall_time_;             //<! The flag to indicate whether to use ros::WallTime or not
+
+  ros::Time last_called_time_;  //!< The last time the keep callback was called
+};
+
+/**
+ * @brief Throttled callback for ROS messages, option to throttle using header timestamps with `message_stamp_callback`
+ *
+ * @tparam M The ROS message type, which should have the M::ConstPtr nested type
+ */
+template <class M>
+class ThrottledMessageCallback : public ThrottledCallback<std::function<void(const typename M::ConstPtr&)>>
+{
+public:
+  using Callback = std::function<void(const typename M::ConstPtr&)>;
+
+  ThrottledMessageCallback(Callback&& keep_callback = nullptr,  // NOLINT(whitespace/operators)
+                           Callback&& drop_callback = nullptr,  // NOLINT(whitespace/operators)
+                           const ros::Duration& throttle_period = ros::Duration(0.0),
+                           const bool use_wall_time = false,
+                           const bool use_header_timestamps = false)
+    : ThrottledCallback<Callback>(std::move(keep_callback), std::move(drop_callback), throttle_period,
+                                  use_wall_time), use_header_timestamps_(use_header_timestamps)
+    {
+      if (use_wall_time && use_header_timestamps) {
+        throw std::runtime_error("Cannot specify both wall time and header timestamps for callback throttling. Use only one.");
```

Comment:
I wonder if it'd make more sense to turn these flags into template parameters :thinking: 

I'm just wondering. It could a separate MR if you think it'd make things cleaner.